desktop/appcontainer/Banner.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2021 Extremely Heavy Industries Inc.
+ */
+import {BannerModel} from '@xh/hoist/appcontainer/BannerModel';
+import {XH, uses, hoistCmp} from '@xh/hoist/core';
+import {div, filler} from '@xh/hoist/cmp/layout';
+import {toolbar} from '@xh/hoist/desktop/cmp/toolbar';
+import {button} from '@xh/hoist/desktop/cmp/button';
+import {Icon} from '@xh/hoist/icon';
+import {isFunction} from 'lodash';
+import classNames from 'classnames';
+
+import './Banner.scss';
+
+/**
+ * Internal component to display a single instance of an app-wide banner.
+ * @see {XH.showBanner()}
+ * @private
+ */
+export const banner = hoistCmp.factory({
+    displayName: 'Banner',
+    model: uses(BannerModel),
+    render({model}) {
+        const {
+            icon,
+            message,
+            intent,
+            className,
+            props
+        } = model;
+
+        return toolbar({
+            className: classNames(
+                'xh-banner',
+                className,
+                intent ? `xh-intent-${intent}` : `xh-intent-none`
+            ),
+            items: [
+                icon,
+                div({
+                    className: 'xh-banner__message',
+                    item: message
+                }),
+                actionButton(),
+                filler(),
+                dismissButton()
+            ],
+            ...props
+        });
+    }
+});
+
+const actionButton = hoistCmp.factory(
+    ({model}) => {
+        const {actionFn, actionButtonProps} = model;
+        if (!isFunction(actionFn)) return null;
+        return button({
+            text: 'Action',
```

Comment:
The lack of a good default text here makes me wonder if we should remove `actionFn` and only take `actionButtonProps` (where user would specify `onClick`). The user almost certainly needs to spec the latter if they are spec'ing the former, why have them split across two top-level configs?

The pattern we have now is from `ErrorMessage`, where we supply a somewhat-more-likely-to-be-useful default of "Retry" for the button text. Weighing in my mind these options:

- Leave everything as it is right now 😃 
- Modify new banner config to accept only `actionButtonProps`
- Modify new banner config + change recently-added `ErrorMessage` comp to also support only the props, removing the default. I doubt many apps have used the comp yet, but we could deprecate/migrate to avoid the breaking change.

Looking at this case here, I'm inclined to go for the last option, but could pretty easily be persuaded to any of them. Let's discuss quick then get on with the merge.

Reply:
I'm happy to implement the last option - makes sense to collapse into one prop. For ErrorMessage, I can't imagine many users are using it, and should be relatively straightforward to deprecate cleanly like you say.

Reply:
Great - let's do it.  I think that makes the most sense, and `actionButtonProps` seems like a reasonable enough config value for the model to trigger the display of an action button.  (Could also go with `actionButton`, but I think the name we have makes it slightly more clear that you should not provide the button element itself.)

Reply:
Great - will make those changes shortly